hal-x86_64/src/serial.rs

```diff
@@ -331,11 +332,6 @@ impl<'a> LockInner<'a> {
         self.inner.is_write_ready()
     }
 
-    #[inline(always)]
-    fn is_read_ready(&self) -> bool {
-        self.inner.is_read_ready()
-    }
```

Comment:
if this is unused i think that's a bug...there's probably someplace where we call `is_write_ready` when we mean to call `is_read_ready`?

Reply:
oh, nvm, we do use it where we're supposed to, it's just that we call the method on `inner`:
https://github.com/hawkw/mycelium/blob/9403955dea67c43b6f7ce28ac270a51aae2df334/hal-x86_64/src/serial.rs#L201
we never use the forwarded impl on the wrapper, so removing this is fine 👍